Tiny js world

Tedzury opened the pull request:
### Tiny_JS_world

[Demo](https://tedzury.github.io/Tiny_JS_world/) |
[Code base](https://github.com/Tedzury/Tiny_JS_world)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

Tedzury commented:
Hello! Fixed using of object.values, replaced it for explicit list of values.  Humbly asking for review :)

OleksiyRudenko commented on submissions/Oleksii_Drohachov/Tiny_JS_world/index.js:
```diff
@@ -0,0 +1,72 @@
+
+   function formOutput(obj) {
+
+      const outputArr = [obj.species, obj.name, obj.gender, obj.legs, obj.hands, obj.saying, obj.friends];
+
+      for (let i = 0; i < outputArr.length; i++) {
+         if (Array.isArray(outputArr[i])) {
+            outputArr[i] = outputArr[i].join(', ')
+         }
+      }
+
+      const result = outputArr.join('; ');
+      
+      return result
```
(1) As a lazy peer developer tasked to add 20 more properties I'd like to those props names without need to write `obj.` every time.

(2) We can `.map` through array of initial data, whatever it is, converting arrays to strings

(3) As after (2) we have only strings and numbers in our array we can chain conversion `.map` with `.join`

(4) ...and finally we can have a single expression and do not need `result` variable.

OleksiyRudenko commented on submissions/Oleksii_Drohachov/Tiny_JS_world/index.js:
```diff
@@ -0,0 +1,72 @@
+   }
+   
+   tinyWorldInhabitants.forEach( item => print(formOutput(item)));
+
+
+
+
```
Tune your IDE to keep one and only one empty line at the end of file.
Also use prettier.io plugin to do code formatting job for you.

OleksiyRudenko reviewed (changes requested):
@Tedzury good job! Let's polish it.
(1) use JS semicolons consistently across your codebase;
(2) see the comments below

Tedzury commented:
Good morning! Tried to fix my code, hope it's fine. 

To be honest, after an hour of tries to get it to work, I cheeted and looked up for finished works of my teemmates in kottans repo. The problem was in the print( ) function. I totally forgot that forEach( ) does not return anything, and desperately tried to return an array for me, to work further, but if I call print inside the forEach, I don't even need the array to form the output, just used join('; ').

Anyway, I'm sorry for that, shame on me :(

OleksiyRudenko commented on submissions/Oleksii_Drohachov/Tiny_JS_world/index.js:
```diff
@@ -0,0 +1,101 @@
+  print(
+    keys
+      .map((key) => {
+        if (Array.isArray(person[key])) {
+          person[key] = person[key].join(", ");
```
This way you change the person's property on spot. What if another fragment of code would like to treat the list differently? We only need this transformation be scoped exclusively for this fragment of code purpose.

You may leverage a ternary operator here. This will also eliminate need of `return` keyword and function body wrapping with `{}`

OleksiyRudenko commented on submissions/Oleksii_Drohachov/Tiny_JS_world/index.js:
```diff
@@ -0,0 +1,101 @@
+    gender: "female",
+    legs: 2,
+    hands: 2,
+    saying: catsPhrase,
```
This approach won't ensure catWoman's saying when cat's will change.
You do not need to fix this here. But you will need different approach when you are back to this JS World after the next task.

OleksiyRudenko reviewed (changes requested):
@Tedzury well done!
Let's polish it a bit.

Tedzury commented:
Good evening! Fixed all issues, changed 'if' for ternary operator, and set a cat as a prototype for catWoman, and for this reason slightly refactored array of characters, to set a proto.

Also want to thank you very much for everything you do for students with that course. I really appreciate that.

OleksiyRudenko reviewed (approved):
@Tedzury job well done!
